Services/User/classes/class.ilUserAppEventListener.php

```diff
@@ -16,5 +16,21 @@ public static function handleEvent($a_component, $a_event, $a_parameter)
                 \ilStartingPoint::onRoleDeleted($a_parameter['object']);
             }
         }
+
+        if ('Services/TermsOfService' === $a_component && ilTermsOfServiceEventWithdrawn::class === $a_event) {
+            global $DIC;
+
+            $shouldDeleteAccountOnWithdrawal = $DIC->settings()->get(
+                'tos_withdrawal_usr_deletion',
+                false
+            );
+
+            $user = new ilObjUser($a_parameter['event']->getUsrId());
+            if($shouldDeleteAccountOnWithdrawal == 1 &&
+                ($user->getAuthMode() == AUTH_LOCAL || $user->getAuthMode() == 'default')
```

Comment:
This is **not** correct. The `default` authentication mode could be `LDAP` (or something else). Please be aware that the `default mode concept` itself will be abandoned AFAIK. As long as this concept still exists, you'll have to check if the `default mode` (`ilSetting`) equals `'local'`.

Why do you try to delete only local accounts here?